comp/core/secrets/component.go

```diff
@@ -18,6 +18,9 @@ type Component interface {
 	Configure(command string, arguments []string, timeout, maxSize int, groupExecPerm, removeLinebreak bool)
 	// Get debug information and write it to the parameter
 	GetDebugInfo(w io.Writer)
-	// Decrypt the given handle and return the corresponding secret value
+	// Decrypt resolves the secrets in the given yaml data by replacing secrets handles by their corresponding secret value
 	Decrypt(data []byte, origin string) ([]byte, error)
+	// DecryptWithCallback resolves the secrets in the given yaml data calling the callback with the YAML path of
+	// the secret handle and its value
+	DecryptWithCallback(data []byte, origin string, callback DecryptCallback) error
```

Comment:
nit: I suppose that the callback is not a "decrypt then run callback" that could be interpreted from current name so maybe wording it as `ResolveViaCallback` or `ResolveThroughCallback` would be more clear

Reply:
We're not resolving using the callback, we're still using the secrets binary. I feel like `ResolveViaCallback`/`ResolveThroughCallback` implies otherwise.

Here we're letting the caller replace the handle with the secrets (ie: resolving with a callback).

Reply:
I did replace `Decrytp` with `Resolve` everywhere.